extras/userdoc/md/documentation/connection-generator-interface.md

```diff
@@ -0,0 +1,55 @@
+# The Connection Generator Interface
+
+Since 2014 ([Djurfeldt et al., 2014](http://dx.doi.org/10.3389/fninf.2014.00043)) NEST supports the Connection Generator Interface. This allows to couple connection generating libraries to NEST without having to modify NEST itself.
+
+In contrast to the [built-in connection functions](connection-management.md) and the Topology module, the Connection Generator Interface has a different way for specifying the connectivity:
+
+2. pre- and postsynaptic neurons are created
+1. a Connection Generator object is created, which specifies the connectivity pattern
+3. the Connection Generator is applied to the sets of source and target neurons using the `CGConnect` function.
+
+## Example in PyNEST using the Connection-Set Algebra
+
+In Python, the `csa` module can be used directly to specify the connectivity.
+
+```python
+import nest
+import csa
+
+# Create the neurons
+sources = nest.Create("iaf_psc_alpha", 2)
+targets = nest.Create("iaf_psc_alpha", 2)
+
+# Create the Connection Generator for one-to-one connectivity
+cg = csa.cset(csa.oneToOne)
+
+# Connect the neurons
+nest.CGConnect(sources, targets, cg)
+
+# Verify the connectivity
+conn = nest.GetConnections(sources)
+print map(lambda x: x["target"], nest.GetStatus(conn))
+```
+
```

Comment:
It would be very useful to have examples showing how to use CG with parameters for the rule (indegree) and initialization of synapse parameters (weight, delay, ...).